pageserver/src/tenant/layer_map/layer_coverage.rs

```diff
@@ -88,14 +88,21 @@ impl<Value: Clone> LayerCoverage<Value> {
             prev_covered = needs_cover;
         }
         if !prev_covered {
-            to_remove.push(key.end);
+            // TODO This line is commented because there's a bug. If uncommented, the
+            //      pageserver_chaos test fails. I suspect it has to do with improper
+            //      handling of key collisions. Will add more tests.
+            // to_remove.push(key.end);
```

Comment:
This could be potentially a really bad bug. It's unrelated to this PR. It's only triggered by the new sorting order. It could potentially exist on main too.